app/src/main/java/com/hedvig/app/feature/home/ui/changeaddress/ChangeAddressViewModel.kt

```diff
@@ -0,0 +1,75 @@
+package com.hedvig.app.feature.home.ui.changeaddress
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.hedvig.app.feature.home.ui.changeaddress.GetSelfChangeEligibilityUseCase.SelfChangeEligibilityResult
+import com.hedvig.app.feature.home.ui.changeaddress.GetUpcomingAgreementUseCase.UpcomingAgreementResult
+import com.hedvig.app.feature.home.ui.changeaddress.ViewState.ChangeAddressInProgress
+import com.hedvig.app.feature.home.ui.changeaddress.ViewState.Loading
+import com.hedvig.app.feature.home.ui.changeaddress.ViewState.ManualChangeAddress
+import com.hedvig.app.feature.home.ui.changeaddress.ViewState.SelfChangeAddress
+import com.hedvig.app.feature.home.ui.changeaddress.ViewState.SelfChangeError
+import com.hedvig.app.feature.home.ui.changeaddress.ViewState.UpcomingAgreementError
+import kotlinx.coroutines.launch
+
+abstract class ChangeAddressViewModel: ViewModel() {
+    abstract val viewState: LiveData<ViewState>
```

Comment:
```suggestion
    protected val _viewState = MutableLiveData<ViewState>()
    val viewState: LiveData<ViewState>
```

using the `protected`-modifier we can provide a publishing-handle for all subclasses as we please. (I used to use this form but switched to what I'm suggesting as I usually always implemented the underlying `MutableLiveData` in both `Impl`s and `Mock`s 🙂